File: javascript/hamming/hamming.js
var Ham = function(){};

Ham.compute = function(strandA, strandB) {
	// The ham number
	var hamCount = 0;

	// If strands do not equal in length we cannot compute
	if ( strandA.length != strandB.length ) {
		throw "DNA strands must be of equal length.";
	}

	// Loop through one string and check them both at i
	for ( var i = 0; i < strandA.length; i++ ) {
		if ( strandA.charAt(i) != strandB.charAt(i) ) {
			hamCount++;
		}
	}

	return hamCount;
};

module.exports = Ham;

File: ruby/hamming/hamming.rb
# Hamming check the difference between 2 strands
class Hamming
  VERSION = 4

  def self.compute(strand_a, strand_b)
    fail ArgumentError unless strand_a.length == strand_b.length

    strand_a.each_char.with_index.count { |_, i| strand_a[i] != strand_b[i] }
  end
end
